src/components/Header/Header.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+type Props = {
+  query: string;
+  setQuery: (query: string) => void;
+  onAdd: () => void;
+  IsDisabled: boolean;
+};
+
+export const Header: React.FC<Props> = React.memo(({
+  query,
+  setQuery,
+  onAdd,
+  IsDisabled,
+}) => {
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className="todoapp__toggle-all active"
+        aria-label="toggle-all"
+      />
+
+      {/* Add a todo on form submit */}
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+
+          onAdd();
+        }}
```

Comment:
In general, it's better to move this function to the separate `handleSubmit` for example